### PR TITLE
[Feature Request][Standalone] Add like expression

### PR DESCRIPTION
**Pull request by winfys**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [x]  Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## Description
like expressions are standard sql expressions, and the main purpose of this pr is to support the like predicate pushdown

## How was this patch tested?
Adds test function to ExpressionSuite.

### CODE REVIEW

File: connectors/standalone/src/main/java/io/delta/standalone/expressions/Like.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package io.delta.standalone.expressions;
+
+/**
+ * Evaluates {@code expr1} like {@code expr2} for {@code new Like(expr1, expr2)}.
+ */
+public class Like extends BinaryOperator implements Predicate {
+    public Like(Expression left, Expression right) {
+        super(left, right, "like");
```
Comment (vkorukanti) [inline connectors/standalone/src/main/java/io/delta/standalone/expressions/Like.java:8]: should we enforce the operand types as `string`?

File: connectors/standalone/src/main/java/io/delta/standalone/expressions/Like.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package io.delta.standalone.expressions;
+
+/**
+ * Evaluates {@code expr1} like {@code expr2} for {@code new Like(expr1, expr2)}.
```
Comment (vkorukanti) [inline connectors/standalone/src/main/java/io/delta/standalone/expressions/Like.java:4]: also add what is the null behavior when the input value is null.

Comment (vkorukanti) [reply connectors/standalone/src/main/java/io/delta/standalone/expressions/Like.java:4]: Also is the `like` functionality same as the SQL Like?

File: connectors/standalone/src/main/java/io/delta/standalone/expressions/Like.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+public class Like extends BinaryOperator implements Predicate {
+    public Like(Expression left, Expression right) {
+        super(left, right, "like");
+    }
+
+    @Override
+    protected Object nullSafeEval(Object leftResult, Object rightResult) {
+        String str = String.valueOf(rightResult);
+        String subStr;
+        if (str.startsWith("%")) {
```
Comment (vkorukanti) [inline connectors/standalone/src/main/java/io/delta/standalone/expressions/Like.java:15]: does this work, if you have `%` or `_` in the middle of the string? Probably it is easier if you change the `%` and `_` to java regex equivalent and just do java regex.

File: connectors/standalone/src/test/scala/io/delta/standalone/internal/ExpressionSuite.scala [hunk 4]
```diff
@@ -594,4 +594,15 @@ class ExpressionSuite extends FunSuite {
...
     assert(lit52.dataType().equals(new DecimalType(5, 2)))
   }
+
+  test("like predicate") {
+    //not like tests
+    testPredicate(new Not(new Like(Literal.of("aabbcc"), Literal.of("aa%"))), false)
+    testPredicate(new Not(new Like(Literal.of("aabbcc"), Literal.of("%aa"))), true)
+    testPredicate(new Not(new Like(Literal.of("aabbcc"), Literal.of("%aa%"))), false)
+    //like tests
+    testPredicate(new Like(Literal.of("aabbcc"), Literal.of("bb%")), false)
```
Comment (vkorukanti) [inline connectors/standalone/src/test/scala/io/delta/standalone/internal/ExpressionSuite.scala:604]: please add few tests where the input is null.

**Review by vkorukanti — COMMENTED** [review]
@scottsand-db Could you please take a look at this PR?

File: connectors/docs/latest/delta-standalone/api/java/io/delta/standalone/expressions/Like.html
Comment (scottsand-db) [inline connectors/docs/latest/delta-standalone/api/java/io/delta/standalone/expressions/Like.html:1]: How did you generate this file?

We usually just run `unidoc` command before each release. We don't need to include this `Like.html` file in this PR.

File: connectors/standalone/src/test/scala/io/delta/standalone/internal/ExpressionSuite.scala [hunk 5]
```diff
@@ -594,4 +594,15 @@ class ExpressionSuite extends FunSuite {
...
+    testPredicate(new Like(Literal.of("aabbcc"), Literal.of("%bb")), false)
+    testPredicate(new Like(Literal.of("aabbcc"), Literal.of("%bb%")), true)
```
Comment (scottsand-db) [inline connectors/standalone/src/test/scala/io/delta/standalone/internal/ExpressionSuite.scala:606]: add tests for: `aa%bb`, `%aa%bb`, `aa%bb%`, `aa%bb%cc`, etc.

File: connectors/standalone/src/main/java/io/delta/standalone/expressions/Like.java (hunk 3 above)
Comment (scottsand-db) [inline connectors/standalone/src/main/java/io/delta/standalone/expressions/Like.java:15]: this implementation seems insufficient to handle multiple `%` usages in the regex string. I think using regex here would be much simpler.

**Review by scottsand-db — CHANGES_REQUESTED** [review]

File: connectors/standalone/src/main/java/io/delta/standalone/expressions/Like.java (hunk 3 above)
Comment (winfys) [reply connectors/standalone/src/main/java/io/delta/standalone/expressions/Like.java]: @scottsand-db @vkorukanti 
LGTM ,  I want to support three matching modes of SQL: like, ilke, and regex, and also support escape characters

**Pull request closed without merging**